### PR TITLE
Snowflake Refactor and add Action

### LINKED ISSUE #973: [ACTION] Snowflake: INSERT new record

### PR DESCRIPTION
**Pull request by andrewjschuang**
- [X] Refactor to ESM
- [X] Insert New Row Action

Resolves #973.

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/snowflake/snowflake.app.mjs [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+export default {
   app: "snowflake",
   type: "app",
+  propDefinitions: {
+    tableName: {
+      type: "string",
+      label: "Table Name",
+      async options(context) {
+        const { page } = context;
+        if (page !== 0) return [];
```
Comment (feyzullah) [inline components/snowflake/snowflake.app.mjs:13]: Please correct me if I am confused. With `if (page !== 0) return []` wouldn't second page always be empty? If we are not going to use `page` variable, can we just delete lines 12 and 13 and return `options` like it is done for `columns` prop?

**Review by feyzullah — COMMENTED** [review]
Hi @andrewjschuang I added just one comment. Can you please check?

**Review by feyzullah — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
@andrewjschuang Observing following error when inserting records. Looks like input column names are not being passed. 
image

**Comment by andrewjschuang** [comment]
> @andrewjschuang Observing following error when inserting records. Looks like input column names are not being passed.

@DilanAthukorala strings must be wrapped in single quotes. I've added this to the description. Weird thing column names not showing, but they are indeed being passed.

**Comment by DilanAthukorala** [comment]
Verification completed

**Review by feyzullah — APPROVED** [review]
Approved

**Pull request merged**